registry/txt.go

```diff
@@ -258,6 +271,9 @@ func (im *TXTRegistry) generateTXTRecord(r *endpoint.Endpoint) []*endpoint.Endpo
 	if txtNew != nil {
 		txtNew.WithSetIdentifier(r.SetIdentifier)
 		txtNew.Labels[endpoint.OwnedRecordLabelKey] = r.DNSName
+		if im.isMigrationEnabled && r.Labels[endpoint.OwnerLabelKey] == im.oldOwnerID {
```

Comment:
Code block not tested
![Screenshot 2025-06-23 at 06 52 28](https://github.com/user-attachments/assets/659f20d0-6f81-4984-b308-54079d418bcd)